### PR TITLE
maturin: 0.9.3 -> 0.9.4

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

Changelog:
https://github.com/PyO3/maturin/blob/v0.9.4/Changelog.md#094---2021-02-18

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
Please do not remove the ofborg internal error even if it succeeded.

**Comment by danieldk** [comment]
> Please do not remove the ofborg internal error even if it succeeded. Someone who manages ofborg probably should check why it failed.

I'd assume that Graham keeps ofborg logs.

**Comment by SuperSandro2000** [comment]
> I'd assume that Graham keeps ofborg logs. After all, these labels are set, so I'd assume that this is also logged. Keeping the label is only confusing for potential reviewers, since the last evaluation did succeed.

He or cole-h told me to not remove those labels so please stick to that. Logs are big and if you don't know what to search for it can be hard.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113670` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aer</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.qiskit-ignis</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python39Packages.qiskit-ibmq-provider: <a href="https://termbin.com/xpwu">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/114bf36857790bb692a854b7009ecd24/raw/index.html">pretty log </a></li>
  </ul>
</details>
<details>
<summary>
41 packages built:
</summary>
<ul>
<li>maturin</li>
<li>python37Packages.fastapi</li>
<li>python37Packages.fastdiff</li>
<li>python37Packages.graphene</li>
<li>python37Packages.johnnycanencrypt</li>
<li>python37Packages.qiskit</li>
<li>python37Packages.qiskit-aer</li>
<li>python37Packages.qiskit-aqua</li>
<li>python37Packages.qiskit-ibmq-provider</li>
<li>python37Packages.qiskit-ignis</li>
<li>python37Packages.qiskit-terra</li>
<li>python37Packages.retworkx</li>
<li>python37Packages.snapshottest</li>
<li>python37Packages.starlette</li>
<li>python37Packages.tumpa</li>
<li>python37Packages.wasmer</li>
<li>python38Packages.fastapi</li>
<li>python38Packages.fastdiff</li>
<li>python38Packages.graphene</li>
<li>python38Packages.johnnycanencrypt</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aer</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-ignis</li>
<li>python38Packages.qiskit-terra</li>
<li>python38Packages.retworkx</li>
<li>python38Packages.snapshottest</li>
<li>python38Packages.starlette</li>
<li>python38Packages.tumpa</li>
<li>python38Packages.wasmer</li>
<li>python39Packages.fastdiff</li>
<li>python39Packages.graphene</li>
<li>python39Packages.johnnycanencrypt</li>
<li>python39Packages.qiskit-terra</li>
<li>python39Packages.retworkx</li>
<li>python39Packages.snapshottest</li>
<li>python39Packages.starlette</li>
<li>python39Packages.tumpa</li>
<li>python39Packages.wasmer</li>
<li>qutebrowser</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.graphene:
<details>


Unused argument: `iso8601`.
Near pkgs/development/python-modules/graphene/default.nix:5:3:

```
  |
5 | , iso8601
  |   ^
```
</details>
python37Packages.johnnycanencrypt:
<details>


`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/build-support/rust/default.nix:259:3:

```
    |
259 |   installPhase = args.installPhase or ''
    |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
python37Packages.qiskit-aer:
<details>


Unused argument: `conan`.
Near pkgs/development/python-modules/qiskit-aer/default.nix:9:3:

```
  |
9 | , conan
  |   ^
```
</details>
python37Packages.tumpa:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python38Packages.graphene:
<details>


Unused argument: `iso8601`.
Near pkgs/development/python-modules/graphene/default.nix:5:3:

```
  |
5 | , iso8601
  |   ^
```
</details>
python38Packages.johnnycanencrypt:
<details>


`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/build-support/rust/default.nix:259:3:

```
    |
259 |   installPhase = args.installPhase or ''
    |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
python38Packages.qiskit-aer:
<details>


Unused argument: `conan`.
Near pkgs/development/python-modules/qiskit-aer/default.nix:9:3:

```
  |
9 | , conan
  |   ^
```
</details>
python38Packages.tumpa:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
python39Packages.graphene:
<details>


Unused argument: `iso8601`.
Near pkgs/development/python-modules/graphene/default.nix:5:3:

```
  |
5 | , iso8601
  |   ^
```
</details>
python39Packages.johnnycanencrypt:
<details>


`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/build-support/rust/default.nix:259:3:

```
    |
259 |   installPhase = args.installPhase or ''
    |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
python39Packages.tumpa:
<details>


Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
</details>

**Comment by danieldk** [comment]
> > I'd assume that Graham keeps ofborg logs. After all, these labels are set, so I'd assume that this is also logged. Keeping the label is only confusing for potential reviewers, since the last evaluation did succeed.
> 
> He or cole-h told me to not remove those labels

Good to know!

**Comment by cole-h** [comment]
Yeah, I requested not to remove the labels because I have a search (https://github.com/NixOS/nixpkgs/pulls?q=label%3Aofborg-internal-error+) that I check every so often (apparently not often enough). Without the label, I'd have to actually watch ofborg logs (or have a similar query, *as well as* one for what actually caused the error).
